torchvision/datasets/celeba.py

```diff
@@ -99,7 +99,12 @@ def __init__(
 
         mask = slice(None) if split_ is None else (splits.data == split_).squeeze()
 
-        self.filename = splits.index
+        if mask == slice(None):  # if split == all
+            self.filename = splits.index
+        elif len(torch.nonzero(mask)) == 1:  # check for the case of a single sample
+            self.filename = [splits.index[torch.nonzero(mask)]]
```

Comment:
How can this branch be triggered? Each split should always have more than one image right? Or is the fake data in the tests setup in a way that one split only has one image? If so, let's change the test data to make it more realistic.

Reply:
Yes this is affected by the fake data, but I thought the more general case would not hurt. But since the more realistic scenario is not having a single sample for any of the splits (specially since the splits are already predetermined by the creators of the dataset), we can just change the fake data for the test case to be more than 1.

Reply:
We should always adapt our tests to fit the implementation and not the other way around. I appreciate the concern for generality, but the dataset is fixed, so if there is nothing triggering this case now, there won't be anything later.

Reply:
I removed the single sample case in the latest commit, and changed the fake data for the 'test' to be 2.